gcloud-java-core/src/main/java/com/google/gcloud/AuthCredentials.java

```diff
@@ -180,22 +234,45 @@ public boolean equals(Object obj) {
     ServiceAccountAuthCredentials(String account, PrivateKey privateKey) {
       this.account = checkNotNull(account);
       this.privateKey = checkNotNull(privateKey);
+      this.credentials = new ServiceAccountCredentials(null, account, privateKey, null, null);
```

Comment:
or `this(new ServiceAccountCredentials(null, account, privateKey, null, null));`
and adding `checkNotNull` to the other constructor `account` and `privateKey` assignments.